### ISSUE TITLE
UAA release tries to create a log directory at a fixed path

### ISSUE THREAD

**Issue by srm09** [cloudfoundry/uaa-release#177]
We deploy UAA as a bosh release along with the PKS application. In the UAA stderr.log file, we see a bunch of these errors showing up.
```
NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
2020-02-14 01:58:21,556 main ERROR FileManager (logs/uaa.log) java.io.IOException: Could not create directory /var/vcap/jobs/uaa/logs java.io.IOException: Could not create directory /var/vcap/jobs/uaa/logs
	at org.apache.logging.log4j.core.util.FileUtils.mkdir(FileUtils.java:127)
	...
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)

2020-02-14 01:58:21,569 main ERROR Could not create plugin of type class org.apache.logging.log4j.core.appender.FileAppender for element File: java.lang.IllegalStateException: ManagerFactory [org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory@5a00eb1e] unable to create manager for [logs/uaa.log] with data [org.apache.logging.log4j.core.appender.FileManager$FactoryData@36fcf6c0] java.lang.IllegalStateException: ManagerFactory [org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory@5a00eb1e] unable to create manager for [logs/uaa.log] with data [org.apache.logging.log4j.core.appender.FileManager$FactoryData@36fcf6c0]
	at org.apache.logging.log4j.core.appender.AbstractManager.getManager(AbstractManager.java:115)
	...
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)

2020-02-14 01:58:21,580 main ERROR Unable to invoke factory method in class org.apache.logging.log4j.core.appender.FileAppender for element File: java.lang.IllegalStateException: No factory method found for class org.apache.logging.log4j.core.appender.FileAppender java.lang.IllegalStateException: No factory method found for class org.apache.logging.log4j.core.appender.FileAppender
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.findFactoryMethod(PluginBuilder.java:234)
	...
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)

2020-02-14 01:58:21,594 main ERROR FileManager (logs/uaa_events.log) java.io.IOException: Could not create directory /var/vcap/jobs/uaa/logs java.io.IOException: Could not create directory /var/vcap/jobs/uaa/logs
	at org.apache.logging.log4j.core.util.FileUtils.mkdir(FileUtils.java:127)
	at org.apache.logging.log4j.core.util.FileUtils.makeParentDirs(FileUtils.java:144)
	...
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)

2020-02-14 01:58:21,609 main ERROR Could not create plugin of type class org.apache.logging.log4j.core.appender.FileAppender for element File: java.lang.IllegalStateException: ManagerFactory [org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory@5a00eb1e] unable to create manager for [logs/uaa_events.log] with data [org.apache.logging.log4j.core.appender.FileManager$FactoryData@39c1fe0b] java.lang.IllegalStateException: ManagerFactory [org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory@5a00eb1e] unable to create manager for [logs/uaa_events.log] with data [org.apache.logging.log4j.core.appender.FileManager$FactoryData@39c1fe0b]
	at org.apache.logging.log4j.core.appender.AbstractManager.getManager(AbstractManager.java:115)
	...
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)

2020-02-14 01:58:21,611 main ERROR Unable to invoke factory method in class org.apache.logging.log4j.core.appender.FileAppender for element File: java.lang.IllegalStateException: No factory method found for class org.apache.logging.log4j.core.appender.FileAppender java.lang.IllegalStateException: No factory method found for class org.apache.logging.log4j.core.appender.FileAppender
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.findFactoryMethod(PluginBuilder.java:234)
	...
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)

2020-02-14 01:58:21,616 main ERROR Null object returned for File in Appenders.
2020-02-14 01:58:21,617 main ERROR Null object returned for File in Appenders.
2020-02-14 01:58:21,617 main ERROR Unable to locate appender "UaaDefaultAppender" for logger config "root"
2020-02-14 01:58:21,622 main ERROR Unable to locate appender "UaaAuditAppender" for logger config "UAA.Audit"
2020-02-14 01:58:21,623 main ERROR Unable to locate appender "UaaDefaultAppender" for logger config "org.cloudfoundry.identity"
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v7.Java7$1 (file:/var/vcap/data/uaa/tomcat/webapps/ROOT/WEB-INF/lib/groovy-2.5.8.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v7.Java7$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
/var/vcap/jobs/uaa/bin/uaa: line 28:    15 Aborted                 /var/vcap/packages/uaa/tomcat/bin/catalina.sh run
```

From the initial looks of it, the logger error is trying to create a file at path `/var/vcap/jobs/uaa/logs`. But the release does not have permissions to create a file under this path, and hence these errors. Eventually we see that the tomcat server process gets aborted.

The contents of the `stdout.log` confirm that the process was restarted after some time.

**Versions affected**: We have seen this issue occur in versions `74.5.5` and `73.4.16`.

**Comment by joshuatcasey** [comment]
Hi @srm09 I'm curious to hear more of your use-case. It's my understanding that `/var/vcap/jobs/uaa` is usually available for a BOSH-deployed UAA. In any case, would you expect one of the following (or something else entirely?):

- Logs go to STDOUT
- Logs to to a location provided via the [spec](jobs/uaa/spec)

**Comment by ionphractal** [comment]
@joshuatcasey
By default UAA, as every other bosh job, should put logs into /var/vcap/**sys**/log/uaa and not under **jobs**. We also get these errors. There's no special use case, but it's probably a bug.

**Comment by shamus** [comment]
Hi @ionphractal agreed, this seems like a bug we'll get it prioritized. It's curious that we haven't received a report about this issue before now. Have you been working around it somehow?

Perhaps this is what @joshuatcasey was getting at?

**Comment by ionphractal** [comment]
@shamus No, it seems the UAA doesn't really care about it. It just errors this in the logs but continues to run. So it wasn't apparent until we ran into another issue that made us check the logs. We didn't work around it.

**Comment by Birdrock** [comment]
Issue has been fixed and released with [v74.15.0](https://github.com/cloudfoundry/uaa-release/releases/tag/v74.15.0)
